From c49713c4d58b411ed15042794b25be24c388460d Mon Sep 17 00:00:00 2001
Message-Id: <c49713c4d58b411ed15042794b25be24c388460d.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 7 Oct 2014 19:54:15 +0530
Subject: [PATCH 15/75] MUST_REBASE [VPG]: drm/i915/adf: Add wrapper for
 display runtime get/put.

This patches adds & exports two interfaces for display
runtime get & put from adf.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Ieef905c6cb76d5506e17d39ed38c8e224f89f5ea
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |   24 ++++++++++++++++++++++++
 include/drm/i915_adf.h                  |    2 ++
 2 files changed, 26 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index e0a32e1..233c291 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -74,6 +74,30 @@ struct pci_dev *i915_adf_get_pci_dev(void)
 }
 EXPORT_SYMBOL(i915_adf_get_pci_dev);
 
+void intel_adf_display_rpm_get(void)
+{
+	struct drm_i915_private *dev_priv;
+
+	if (!i915_adf_dev)
+		return;
+
+	dev_priv = i915_adf_dev;
+	intel_display_set_init_power(dev_priv, true);
+}
+EXPORT_SYMBOL(intel_adf_display_rpm_get);
+
+void intel_adf_display_rpm_put(void)
+{
+	struct drm_i915_private *dev_priv;
+
+	if (!i915_adf_dev)
+		return;
+
+	dev_priv = i915_adf_dev;
+	intel_display_set_init_power(dev_priv, false);
+}
+EXPORT_SYMBOL(intel_adf_display_rpm_put);
+
 /**
  * intel_adf_pci_sideband_rw - Interface to allow ADF driver read/write to intel sideband.
  */
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index fe3a0e1..40bc75c 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -53,6 +53,8 @@ extern u32 intel_adf_get_pwm_vbt_data(void);
 extern void set_adf_ready(void);
 extern volatile bool g_adf_ready;
 extern int intel_adf_context_on_event(void);
+extern void intel_adf_display_rpm_get(void);
+extern void intel_adf_display_rpm_put(void);
 #endif
 
 
-- 
1.7.9.5

